### DISCUSSION TITLE
Indicator/Strategy alerts

### DISCUSSION THREAD

**Discussion by nkarabulut** [Mathieu2301/TradingView-API#75] [Q&A] [answered]
Hello,

Thank you for developing this package. But I need a little help. (Suggestion or sniplett)

We can get the indicator data which is great.

I want to receive the indicator alert data via websocket client. (I don't want to use webhook)

const client = new TradingView.Client(); //connection ok

The client object has the onData function, but as output;
{session_id,timestamp,timestampMs,release,studies_metadata_hash,protocol,javastudies,auth_scheme_vsn,via}
json containing information is returned. I think the indicator data comes as a stream over websocket. It must be sending alert information over websocket. How can I access this stream? Even if raw data comes, I can try to parse it.

In short:
I want to receive alert data with websocket client. Can you suggest a way?

Thanks.

**Comment by Mathieu2301** [comment, marked as answer]
Hello,

So you are talking about Alerts. 
<img width="243" alt="image" src="https://user-images.githubusercontent.com/21021423/155687957-814958f9-31ad-4303-be2d-d62241ca2075.png">

It turns out that the alert system does not go through websockets. There are no plans to implement the alert system in the library.

I suggest you to use the `getDrawings` function. Then you check if the price (or indicator plot value) crosses a line.

**Reply by nkarabulut** [reply]
Thanks for your reply.

I will try to get the indicator from plot variables.

It may sound silly, but I can't get session id

Error: Wrong or expired sessionid

I'm trying to get it from the browser's session store, I couldn't figure it out, I couldn't proceed
<img width="853" alt="Screen Shot 2022-02-25 at 16 15 42" src="https://user-images.githubusercontent.com/6299528/155721992-5a4a1cbe-2003-4388-9037-acb7d7493978.png">
.

Do we get session id differently?

Thanks..

**Comment by nkarabulut** [comment]
I try in safari and chrome

**Comment by neelgo** [comment]
Suggestion:
If it is a strategy not study then may be you can use
console.log('Study strategyReport:', myIndicator.strategyReport.trades[0]);
this will give you most recent trade
you need to read entry record. and then exit record.
I hope your strategy not firing too fast otherwise you will see both entry and exit together.
